src/OpenDebugAD7/TextPositionTuple.cs

```diff
@@ -41,6 +41,12 @@ public static TextPositionTuple GetTextPositionOfFrame(PathConverter converter,
                     Path = convertedFilePath,
                     Name = Path.GetFileName(convertedFilePath)
                 };
+                if (!File.Exists(convertedFilePath))
```

Comment:
This seems like a reasonable approach to me, but I think you want to move this to HandleStackTraceRequestAsync. The other caller of this function (FireStoppedEvent) will not care about this.

It is probably also worth doing a parrellel for (or similar) over all the frames rather than doing this synchronously, and using a timeout since hitting random parts of the file system can hang indefinitely.

Reply:
Well in theory it does but you are right, in practice vscode does not use the source property of the stopped event.

Reply:
Suggestions welcome for how to do the timeout thing.

Reply:
I thought `Parrellel.ForEach` had that option, but I guess I was wrong. I think you would need to instead do:

```
        protected override async void HandleStackTraceRequestAsync(...)
        {
            try
            {
                ...

                CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
                try
                {
                    CancellationToken cancellationToken = cancellationTokenSource.Token;
                    Task[] frameTasks = inputs.Select(x => Task.Run(() =>
                    {
                        cancellationToken.ThrowIfCancellationRequested();
                        ...
                    })).ToArray();

                    Task allFramesComplete = Task.WhenAll(frameTasks);
                    Task timeoutTask = Task.Delay(20000, cancellationToken);
                    try
                    {
                        await Task.WhenAny(allFramesComplete, timeoutTask);
                    }
                    catch (Exception e)
                    {
                        Debug.Fail("Ignoring exception computing frame state: " + e.ToString());
                    }
                }
                finally
                {
                    cancellationTokenSource.Cancel();
                    cancellationTokenSource.Dispose();
                }
            }
            catch(Exception e)
            {
                responder.SetError(...);
            }
        }            
```

Reply:
Not sure if it's worth it though since it's gonna be superseded by #1041.

Reply:
I am not sure I understand what you mean - I think in general if we are going to add File.Exists checks on each frame we should do something like I have.

Reply:
Yeah true, so in #1041 we're mostly interested in the lowest frame but I check the other ones too to mark them.

Reply:
Seems awfully complicated though just to handle an unlikely(?) event.

Reply:
Please check my update.